prottp.go

```diff
@@ -151,39 +185,41 @@ func wrapMethod(service interface{}, m grpc.MethodDesc) http.Handler {
 			}
 			return
 		}
-		writeMessage(isJson, resp.(proto.Message), http.StatusOK, w)
+		WriteMessage(statusCode, resp.(proto.Message), w, r)
 	})
 }
 
-func writeMessage(isJson bool, msg proto.Message, statusCode int, w http.ResponseWriter) {
+// WriteMessage is exported to be used for middleware to return proto message
+func WriteMessage(statusCode int, msg proto.Message, w http.ResponseWriter, r *http.Request) {
 	var err error
+	var isJSON = isMimeTypeJSON(w.Header().Get("Content-Type"))
+	if w.Header().Get("Content-Type") == "" {
+		isJSON = shouldReturnJSON(r)
+		if isJSON {
+			w.Header().Set("Content-Type", fmt.Sprintf("%s;type=%s", jsonContentType, proto.MessageName(msg)))
+		} else {
+			w.Header().Set("Content-Type", fmt.Sprintf("%s;type=%s", xprottpContentType, proto.MessageName(msg)))
```

Comment:
Minimise the code inside the conditionals from "set the content type header to the content type and the name of the message" to the actual difference: "figure out the content type", then below that, do the common work:

```diff
-if isJSON {
-	w.Header().Set("Content-Type", fmt.Sprintf("%s;type=%s", jsonContentType, proto.MessageName(msg)))
-} else {
-	w.Header().Set("Content-Type", fmt.Sprintf("%s;type=%s", xprottpContentType, proto.MessageName(msg)))
-}
+contentType := xprottpContentType
+if isJSON {
+	contentType = jsonContentType
+}
+w.Header().Set("Content-Type", fmt.Sprintf("%s;type=%s", contentType, proto.MessageName(msg)))
```

Reply:
will fix this.